ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ProfileStatsWidget.tsx

```diff
@@ -0,0 +1,129 @@
+import { Chart, ChartAxis, ChartBar, ChartContainer, ChartLabel } from '@patternfly/react-charts';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+import EmptyStateTemplate from 'Components/EmptyStateTemplate';
+import useResizeObserver from 'hooks/useResizeObserver';
+import { ComplianceProfileScanStats } from 'services/ComplianceResultsStatsService';
+import { defaultChartHeight, defaultChartBarWidth } from 'utils/chartUtils';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+import {
+    FAILING_COLOR,
+    MANUAL_COLOR,
+    OTHER_COLOR,
+    PASSING_COLOR,
+} from '../compliance.coverage.constants';
+import { getStatusCounts } from '../compliance.coverage.utils';
+
+type ChartData = {
+    x: string;
+    y: number;
+    color: string;
+};
+
+type DatumArgs = {
+    datum?: ChartData;
+};
+
+export type ProfileStatsWidgetProps = {
+    isLoading: boolean;
+    error: Error | undefined;
+    profileScanStats: ComplianceProfileScanStats | undefined;
+};
+
+function ProfileStatsWidget({ error, isLoading, profileScanStats }: ProfileStatsWidgetProps) {
+    const [widgetContainer, setWidgetContainer] = useState<HTMLDivElement | null>(null);
+    const widgetContainerResizeEntry = useResizeObserver(widgetContainer);
+
+    if (isLoading) {
+        return (
+            <Bullseye>
+                <Spinner />
+            </Bullseye>
+        );
+    }
+
+    if (error) {
+        return (
+            <Bullseye>
+                <EmptyStateTemplate
+                    title="Error loading profile stats"
+                    headingLevel="h3"
+                    icon={ExclamationCircleIcon}
+                    iconClassName="pf-v5-u-danger-color-100"
+                >
+                    {getAxiosErrorMessage(error.message)}
+                </EmptyStateTemplate>
+            </Bullseye>
+        );
+    }
+
+    if (profileScanStats) {
+        const { passCount, failCount, manualCount, otherCount, totalCount } = getStatusCounts(
+            profileScanStats.checkStats
+        );
+
+        const data: ChartData[] = [
+            {
+                x: 'Passing',
+                y: passCount / totalCount,
+                color: PASSING_COLOR,
+            },
+            {
+                x: 'Failing',
+                y: failCount / totalCount,
+                color: FAILING_COLOR,
+            },
+            {
+                x: 'Manual',
+                y: manualCount / totalCount,
+                color: MANUAL_COLOR,
+            },
+            {
+                x: 'Other',
+                y: otherCount / totalCount,
+                color: OTHER_COLOR,
+            },
+        ];
+
+        return (
+            <div ref={setWidgetContainer}>
+                <Chart
+                    ariaDesc="Aging images grouped by date of last update"
+                    ariaTitle="Aging images"
```

Comment:
Are these copy paste? Bravo for applying existing patterns!

Reply:
Whoops